Adding aliquot sum algorithm

aidnyu opened the pull request:
This PR introduces a simple aliquot sum algorithm in mojo, with doc tests.

- [x] I have performed a self-review of my code
- [x] My code follows the style guidelines of this project
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Comments in areas I changed are up to date
- [x] I have added comments to hard-to-understand areas of my code

SatinWukerORIG reviewed (changes requested):
The code looks great! However, it would be better if you can add a little example in the doc comment, like: "for example, the aliquot sum of 12 is 1 + 2 + 3 + 4 + 6 =16"

aidnyu commented:
@SatinWukerORIG I've expanded the documentation with multiple detailed examples:

- Added explanation of what proper divisors are
- Included examples for various types of numbers:
  * Composite number (12): showing all divisors and their sum
  * Perfect number (6): where sum equals the number itself
  * Another composite number (15): for completeness
  * Prime number (19): showing special case with only 1 as divisor
  * The number 1: showing edge case with zero sum

Let me know if you'd like any other examples or clarifications!

SatinWukerORIG reviewed (commented):
The doc looks good to me 👍
But I think it is possible to implement this algorithm in O(sqrt(n)):
```python
 for i in range(2, sqrt_n + 1):
    if n % i == 0:
        total += i
        if i != n // i:
            total += n // i
    return total
```
When a divisor i is found, its pair n // i is also a divisor. That means you only need to loop over sqrt(n), right?

aidnyu commented:
> The doc looks good to me 👍
> But I think it is possible to implement this algorithm in O(sqrt(n)):
> ```python
>  for i in range(2, sqrt_n + 1):
>     if n % i == 0:
>         total += i
>         if i != n // i:
>             total += n // i
> 
> # at the end of the function:
> return total
> ```
> When a divisor i is found, its pair n // i is also a divisor. That means you only need to loop over sqrt(n), right?
> 
> 

@SatinWukerORIG Thank you for the suggestion! 

I was aware of the O(sqrt(n)) optimization, but chose the simpler approach initially as this repository serves educational purposes. The straightforward implementation helps beginners understand the core concept more easily. 

Perhaps we could add this optimization as a separate function or with comments explaining both approaches? This would provide educational value by showing algorithm evolution from basic to optimized implementations.​​​​​​​​​​​​​​​​

SatinWukerORIG commented:
@aidnyu 
That's a good point!
It is much better if we can provide two implementations to lead learners to jump from the basic implementation to the optimized one. You can add that.

aidnyu commented:
@SatinWukerORIG Thank you! 

I've implemented both versions and used a clearer naming convention with `optimized_aliquot_sum` for the O(sqrt(n)) implementation.

aidnyu commented:
@reviewer Thank you! 

I've implemented both versions as you suggested. 

The main aliquot_sum function maintains the basic O(n) implementation which is more intuitive for learners, and I've added an optimized O(sqrt(n)) version that leverages the divisor pairs approach.
